src/charm.py

```diff
@@ -418,6 +416,30 @@ def _report_service_ips(self):
             x.spec.cluster_ip for x in services.items if x.metadata.name in all_k8s_service_names
         ]
 
+    def _report_ingress_ips(self) -> str:
+        """Report on ingress IP(s)."""
+        self.k8s_auth()
+        api = _networking_v1_api()
+
+        # Wait up to `interval * count` seconds for ingress IPs.
+        count, interval = 100, 1
+        for _ in range(count):
+            ingresses = api.list_namespaced_ingress(namespace=self._namespace)
+            # XXX: Will this return multiple IPs if there are any?
+            try:
+                ips = [x.status.load_balancer.ingress[0].ip for x in ingresses.items]
+            except TypeError:
+                # We have no IPs yet.
+                ips = []
+            if ips:
+                break
+            LOGGER.info("Sleeping for %s seconds to wait for ingress IP", interval)
+            time.sleep(interval)
+        if ips:
+            return "Ingress IP(s): {}, ".format(str(ips))
```

Comment:
This part should move to building the message, and we should use a join, similar to how we currently do for service IPs.